packages/backend/src/apps/toolbox/common/get-branch-step-id-to-skip-to.ts

```diff
@@ -0,0 +1,51 @@
+import type { IGlobalVariable, IStep } from '@plumber/types'
+
+import Step from '@/models/step'
+
+export async function getBranchStepIdToSkipTo(
+  $: IGlobalVariable,
+): Promise<IStep['id']> {
+  // PERF-FIXME: Objectionjs does no caching, so this will almost always be
+  // queried multiple times by the same worker during a test run. If it does
+  // turn out to impact perf, we can LRU memoize this by executionId.
+  const flowSteps = await Step.query()
+    .where('flow_id', $.flow.id)
+    .orderBy('position', 'asc')
+    .throwIfNotFound()
+
+  // search for immediate branch before current step
+  const currBranchStep = flowSteps
+    .slice(0, $.step.position) // position is 1-indexed so if-then takes current step by default
+    .reverse()
```

Comment:
a bit scary since .reverse() is in-place but luckily there's a slice before this that creates a copy before reverse()

Reply:
yep if not i would have created the array outside